### ISSUE TITLE
Text2Interaction training loss becomes NaN after ~2.6k steps

### ISSUE THREAD

**Issue by wenqi-cai297** [wzyabcas/InterAct#33]
```markdown
Hi, thanks for releasing the InterAct code and data.

We are trying to reproduce the Text2Interaction training following the README instructions, but the training run becomes NaN early. Below are the environment, command, and observed loss curve. Could you help us understand why this happens and how to resolve it?

### Environment

- GPU: NVIDIA GeForce RTX 5080, 16GB
- Driver: 595.71.05
- `nvidia-smi` CUDA version: 13.2
- Python: 3.10.20
- PyTorch: `2.7.1+cu128`
- `torch.version.cuda`: 12.8
- Single GPU training with `CUDA_VISIBLE_DEVICES=2`
- `pointnet2_ops` import works after compiling with:
  - `CUDA_HOME=$CONDA_PREFIX`
  - `TORCH_CUDA_ARCH_LIST=12.0`

### Codebase

We use the released InterAct repository. All paths below are relative to the InterAct repository root.

```bash
cd <INTERACT_REPO_ROOT>/text2interaction
```

The README training command appears to be:

```bash
python -m train.hoi_diff --save_dir ./save/t2m_interact --dataset interact
```

### Data loading

The full `interact` dataset loads successfully. The training log shows:

```text
Loading dataset ['neuraldome', 'chairs', 'omomo', 'imhd', 'grab', 'behave', 'intercap'] ...
Loading dataset neuraldome ... 1491
Loading dataset chairs ... 1502
Loading dataset omomo ... 4294
Loading dataset imhd ... 595
Loading dataset grab ... 1347
Loading dataset behave ... 1766
Loading dataset intercap ... 168
```

With `batch_size=64`, the training epoch has 148 iterations.

### Exact training command

```bash
cd <INTERACT_REPO_ROOT>/text2interaction

export PYTHONPATH=.
export CUDA_HOME="$CONDA_PREFIX"
export TORCH_CUDA_ARCH_LIST="12.0"
export CPATH="$CONDA_PREFIX/targets/x86_64-linux/include:$CONDA_PREFIX/include:${CPATH:-}"
export LIBRARY_PATH="$CONDA_PREFIX/targets/x86_64-linux/lib:$CONDA_PREFIX/lib:${LIBRARY_PATH:-}"
export LD_LIBRARY_PATH="$CONDA_PREFIX/targets/x86_64-linux/lib:$CONDA_PREFIX/lib:${LD_LIBRARY_PATH:-}"
export TORCH_EXTENSIONS_DIR="<TORCH_EXTENSIONS_DIR>"
mkdir -p "$TORCH_EXTENSIONS_DIR"

CUDA_VISIBLE_DEVICES=2 \
python -m train.hoi_diff \
  --save_dir ./save/repro_nan_bs64_lr1e4 \
  --dataset interact \
  --device 0 \
  --batch_size 64 \
  --train_platform_type TensorboardPlatform \
  --log_interval 100 \
  --save_interval 10000 \
  --num_steps 200000 \
  --lr 1e-4 \
  --weight_decay 0.0 \
  --overwrite
```

### Result

The loss decreases initially, then starts increasing, and becomes NaN between step 2500 and 2600:

```text
step[0]: loss[1.85546]
step[100]: loss[0.55678]
step[200]: loss[0.44338]
step[300]: loss[0.37129]
step[400]: loss[0.32432]
step[500]: loss[0.29076]
step[600]: loss[0.26600]
step[700]: loss[0.24671]
step[800]: loss[0.23110]
step[900]: loss[0.21794]
step[1000]: loss[0.20696]
step[1100]: loss[0.19767]
step[1200]: loss[0.18930]
step[1300]: loss[0.18240]
step[1400]: loss[0.20120]
step[1500]: loss[0.22500]
step[1600]: loss[0.24487]
step[1700]: loss[0.26188]
step[1800]: loss[0.27663]
step[1900]: loss[0.28999]
step[2000]: loss[0.30186]
step[2100]: loss[0.31252]
step[2200]: loss[0.32206]
step[2300]: loss[0.33071]
step[2400]: loss[0.33862]
step[2500]: loss[0.34603]
step[2600]: loss[nan]
```

After step 2600, all subsequent logged losses remain NaN.

### Questions

1. Does this command and environment look consistent with the intended Text2Interaction training setup?
2. Do you know why the loss starts increasing after about step 1300 and becomes NaN around step 2600?
3. What would you recommend for resolving this NaN issue?
4. Are there any known data samples, masks, or training settings that can trigger this behavior? We noticed `masked_l2()` divides by `non_zero_elements = sum_flat(mask) * n_entries` without a zero guard.

Any guidance would be very helpful. We are happy to provide the full log or run additional diagnostics if needed.
```

## Local Evidence Checked

- `README.md` documents the Text2Interaction training command as `python -m train.hoi_diff --save_dir ./save/t2m_interact --dataset interact`.
- `text2interaction/model/mdm.py` loads and freezes the TMA text encoder from `assets/eval/markersbps.ckpt`.
- `text2interaction/diffusion/gaussian_diffusion.py` implements `masked_l2()` as `loss / non_zero_elements`, without a zero guard.
